shell/common/rasterizer.cc

```diff
@@ -364,6 +382,19 @@ void Rasterizer::SetResourceCacheMaxBytes(int max_bytes) {
   }
 }
 
+std::optional<size_t> Rasterizer::GetResourceCacheMaxBytes() const {
+  if (!surface_) {
```

Comment:
The validity of this call still depends on the the presence of a surface. Why not just access the optional and return the size_t instead of an optional? Now that the value is maintained between teardowns and setups, the contexts cache size should always be consistent with that in the optional.

Reply:
What happens if this is called before the first setup, and before anyone has attempted to set the metrics?

Reply:
Return `kGrCacheMaxByteSize` maybe?

Reply:
But that's not really accurate, and that symbol is defined as an implementation detail of gl_surface_cpu.cc.  It's different on Fuchsia for instance, where they override it in another place.

Reply:
Ok then. Bad idea. Thanks for clarifying.